internal/engine/buildcontroller.go

```diff
@@ -78,6 +83,50 @@ func nextTargetToBuild(state store.EngineState) *store.ManifestTarget {
 	return choice
 }
 
+type noBuildManifestsFirst struct {
+	mts             []*store.ManifestTarget
+	origIndexByName map[string]int
```

Comment:
I'm surprised you didn't make the key a ManifestName

Reply:
me too!